tablestore/database.go

```diff
@@ -0,0 +1,241 @@
+package tablestore
+
+import (
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"github.com/google/uuid"
+	"sync"
+	"fmt"
+	//"strconv"
+	"io"
+	"bytes"
+)
+
+// StateMachine interface provides execution of command against database
+// the implementation should be thread safe
+type Store interface {
+	//Execute(c Command, ID ids.ID) (Value, error)
+	Save(tableid uuid.UUID, items []KVItem) error
+	GetPartition(tableid uuid.UUID, partitionKeys []FleetDBValue, columns []string,  tableSpec FleetDBTableSpec) ([]Row, error)
+}
+
+// database maintains the key-value datastore
+type database struct {
+	lock 		*sync.RWMutex
+	leveldbs    map[uuid.UUID] *leveldb.DB  // map of leveldb shards
+}
+
+// NewStore get the instance of LevelDB Wrapper
+func NewStore() Store {
+	//log.Infof("Starting KV-store at node %v \n", ids.GetID())
+	db := new(database)
+	db.lock = new(sync.RWMutex)
+	db.leveldbs = make(map[uuid.UUID] *leveldb.DB)
+	return db
+}
+
+func (db *database) getStore(tableid uuid.UUID) *leveldb.DB{
+	lvldb := db.leveldbs[tableid]
+	dir := "/tmp/lvldb" 
+	if lvldb == nil {
+		//storedbs = make([]*leveldb.DB, 0)
+		//for i, dir := range dirs {
+			lvlDBName := dir + "/" + tableid.String()
+			lvldb, _ = leveldb.OpenFile(lvlDBName,nil)
+			//storedbs = append(storedbs, lvldb)
+			//if err == nil {
+			//	fmt.Println(err)
+			//}
+		//}
+		db.leveldbs[tableid] = lvldb;
+	}
+	return lvldb
+
+}
+
+func (db *database) Save(tableid uuid.UUID, items []KVItem) error {
+   //Handle put, update, delete
+    storedb := db.getStore(tableid)
+    var err error
+    //for _ ,currleveldb := range storedbs{
+    	batch := new(leveldb.Batch)
+    	for _, item := range items{
+    		if(item.Value == nil){
+    			batch.Delete(item.Key)
+    		}else{
+    			batch.Put(item.Key,item.Value)
+    		}
+    	}
+    	//batch.Put([]byte("foo"), []byte("value"))
```

Comment:
clean up the commented code. If something needs to be commented, please describe it in English and use example code if needed. But do not leave just lines and lines of commented code